stubs/openpyxl/openpyxl/worksheet/_reader.pyi

```diff
@@ -46,15 +49,15 @@ class WorkSheetParser:
     min_row: Incomplete | None
     min_col: Incomplete | None
     epoch: datetime
-    source: Incomplete
-    shared_strings: Incomplete
+    source: _FileRead
+    shared_strings: SupportsGetItem[int, Incomplete]
```

Comment:
It doesn't feel right to have such a simple Protocol as an attribute, but openpyxl's own usage of it is extremely simple. I don't see the property being dynamically bound anywhere else. It's in a private module. And it's undocumented.